Purchases/Purchasing/StoreKitAbstractions/SK2StoreProduct.swift

```diff
@@ -60,14 +66,10 @@ internal struct SK2StoreProduct: StoreProductType {
 
     var priceFormatter: NumberFormatter? {
         guard let currencyCode = self.currencyCode else {
-          Logger.appleError("Can't initialize priceFormatter for SK2 product! Could not find the currency code")
-          return nil
-      }
-        let formatter = NumberFormatter()
-        formatter.numberStyle = .currency
-        formatter.currencyCode = currencyCode
-        formatter.locale = .autoupdatingCurrent
-        return formatter
+            Logger.appleError("Can't initialize priceFormatter for SK2 product! Could not find the currency code")
+            return nil
+        }
+        return priceFormatterProvider.priceFormatterForSK2(withCurrencyCode: currencyCode)
```

Comment:
I love that at the very least all this code is now abstracted out of these 3 types!